### PR TITLE
add pre-commit hook to enforce notebooks run order

### PR DESCRIPTION
**Pull request by AgnieszkaZaba**
Solution to  #27 as pre-commit hook

### CODE REVIEW

**Comment by AgnieszkaZaba** [comment]
For resolving #16 we need to add another hook to check if at least one output isn't an empty list.

File: test_notebooks.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ def _repo_path():
     name="notebook_filename",
 )
 def _notebook_filename(request):
-    return request.param
+    return os.path.normpath(request.param)
```
Comment (slayoo) [inline test_notebooks.py:60]: Thanks!
Is this change intentional?

File: test_files/template.ipynb
Comment (slayoo) [inline test_files/template.ipynb:1]: does the `enforce-notebook-run-order` action also enforce presence of output?

File: test_notebooks.py (hunk 1 above)
Comment (AgnieszkaZaba) [reply test_notebooks.py:60]: Oh definitely not. I think I did something wrong when merging main to my branch az-pre-commit

Comment (AgnieszkaZaba) [reply test_notebooks.py]: > Should I change it back with new commit?

I changed it back.

**Comment by slayoo** [comment]
BTW, this PR contains 27 commits as of now, most of which are unrelated. It is because the branch was branched off a previous PR's branch, instead of from `main` branch. It can be resolved by rebasing this PR's branch to the current `main`.

File: test_files/template.ipynb
Comment (slayoo) [reply test_files/template.ipynb:1]: So, isn't it rather "no" then "yes"? :)

**Review by slayoo — DISMISSED** [review]

**Comment by slayoo** [comment]
@AgnieszkaZaba, this PR is of course OK, but it actually does not help us with maintenance of PySDM, PyPartMC or PyMPDATA - the pre-commit hook would need to be added in these repos, not here. I wonder if creating a pytest test to check for cell execution order wouldn't still be worth adding?

File: test_files/template.ipynb
Comment (AgnieszkaZaba) [reply test_files/template.ipynb:1]: Yes and no :) there is a very small difference between not having an output and having empty one. After executing a cell, new key "outputs" is added to that cell and it contains empty list.

Comment (slayoo) [reply test_files/template.ipynb:1]: Empty output is of course perfectly OK, the point is to enforce that the committed notebooks were executed

**Comment by AgnieszkaZaba** [comment]
> @AgnieszkaZaba, this PR is of course OK, but it actually does not help us with maintenance of PySDM, PyPartMC or PyMPDATA - the pre-commit hook would need to be added in these repos, not here. I wonder if creating a pytest test to check for cell execution order wouldn't still be worth adding?

Probably it would be better and easier to add check for output.

**Comment by AgnieszkaZaba** [comment]
> BTW, this PR contains 27 commits as of now, most of which are unrelated. It is because the branch was branched off a previous PR's branch, instead of from `main` branch. It can be resolved by rebasing this PR's branch to the current `main`.

@slayoo I think I did something wrong here

**Comment by slayoo** [comment]
> > BTW, this PR contains 27 commits as of now, most of which are unrelated. It is because the branch was branched off a previous PR's branch, instead of from `main` branch. It can be resolved by rebasing this PR's branch to the current `main`.
> 
> @slayoo I think I did something wrong here

reverted. let's merge and keep the issue open
Thanks!

**Review by slayoo — APPROVED** [review]

**Comment by AgnieszkaZaba** [comment]
> > > BTW, this PR contains 27 commits as of now, most of which are unrelated. It is because the branch was branched off a previous PR's branch, instead of from `main` branch. It can be resolved by rebasing this PR's branch to the current `main`.
> > 
> > 
> > @slayoo I think I did something wrong here
> 
> reverted. let's merge and keep the issue open Thanks!

Thanks!
